autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -96,24 +105,27 @@ def _fit(
         # The final output of a validator is a numpy array. But pandas
         # gives us information about the column dtype
         if isinstance(X, np.ndarray):
+
             X = self.numpy_array_to_pandas(X)
+            # Replace the data type from the previously saved type.
+            self.data_type = type(X)
+            # save all the information about the column order and data types
+            self._check_data(X)
 
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
-            X = cast(pd.DataFrame, X)
-
-            if not X.select_dtypes(include='object').empty:
-                X = self.infer_objects(X)
 
-            self._check_data(X)
```

Comment:
should we not have a check data for pandas data frame?

Reply:
To my understanding no, because it is already checked in `fit` at the `base_feature_validator`, here:

https://github.com/automl/Auto-PyTorch/blob/217c38dd43b012d3839f757cce7877299aac74cc/autoPyTorch/data/base_feature_validator.py#L93

Reply:
also set the self.data_type for both pandas?

Reply:
With what I saw no, because it is already set once when calling `_check_data` from fit in the `base_feature_validator`. The type does not change, it is still a DataFrame.

Reply:
oh ohkay I forgot about that. Thanks